core/src/datayoga_core/blocks/relational/write/block.py

```diff
@@ -162,6 +166,43 @@ def generate_upsert_stmt(self) -> Any:
                 ", ".join([f"target.{column} = {sa.bindparam(column)}" for column in self.mapping_columns])
             ))
 
+    def process_records(
+        self,
+        records: List[Dict[str, Any]],
+        execute_method: Callable[[List[Dict[str, Any]]], None]
+    ) -> Tuple[List[Result], List[Result]]:
+        """Processes records using the given execute method.
+
+        Args:
+            records (List[Dict[str, Any]]): List of records to process.
+            execute_method (Callable[[List[Dict[str, Any]]], None]) Method to execute records (e.g., execute_upsert or execute_delete).
+
+        Returns:
+            Tuple[List[Result], List[Result]]: Processed and rejected records.
+        """
+        processed_records: List[Result] = []
+        rejected_records: List[Result] = []
+
+        try:
+            execute_method(records)
+            processed_records.extend([Result(Status.SUCCESS, payload=record) for record in records])
+        except Exception as batch_error:
+            logger.warning(f"Batch operation failed: {batch_error}")
```

Comment:
add "will be retried one by one" or something to indicate it is a warning and not an error

Reply:
done